subsys/llext/llext_link.c

```diff
@@ -435,7 +435,7 @@ int llext_link(struct llext_loader *ldr, struct llext *ext, const struct llext_l
 				 */
 				link_addr = (uintptr_t)llext_loaded_sect_ptr(ldr, ext,
 									     sym.st_shndx)
-					    + sym.st_value;
+					    + sym.st_value + rel.r_addend;
```

Comment:
`.r_addend` is defined only for RELA type relocations, whereas AFAICS this code can also handle REL type ones, which don't have `.r_addend` but just use the same structure type for convenience. The `rel` variable is on stack and in the REL case its `.r_addend` field will stay uninitialised and will contain random data. You can easily fix this by initialising the variable as `elf_rela_t rel = {};` or even explicitly `elf_rela_t rel = {.r_addend = 0,};`

Reply:
Also, this addition is relocation-specific: some ops do expect an offset (either in `RELA` structs or in place of the opcode for `REL` entries), some do not. Moving this to the ARC-specific `elf.c` should fix the CI issues.

Reply:
Moved into ARC elf.c